firefox_puppeteer/tests/test_toolbars.py

```diff
@@ -0,0 +1,132 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import Wait
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+from firefox_ui_harness.decorators import skip_under_xvfb
+
+
+class TestLocationBar(FirefoxTestCase):
+
+    def test_reload(self):
+        event_types = ["shortcut", "shortcut2", "button"]
+        for event in event_types:
+            # TODO: Until we have waitForPageLoad, this only tests API
+            # compatibility.
+            self.browser.navbar.locationbar.reload_url(event, True)
+            self.browser.navbar.locationbar.reload_url(event, False)
+
+    def test_focus_and_clear(self):
+        locationbar = self.browser.navbar.locationbar
+        locationbar.urlbar.send_keys("zyx")
+        locationbar.clear()
+        self.assertEqual(locationbar.value, '')
+        locationbar.urlbar.send_keys("zyx")
+        self.assertEqual(locationbar.value, 'zyx')
+        locationbar.clear()
+        self.assertEqual(locationbar.value, '')
+
+    def test_load_url(self):
+        data_uri = 'data:text/html,<title>Title</title>'
+        locationbar = self.browser.navbar.locationbar
+        locationbar.clear()
+        locationbar.load_url(data_uri)
+
+        with self.marionette.using_context('content'):
+            Wait(self.marionette).until(
+                lambda mn: mn.get_url() == data_uri)
+
+    def test_urlbar_input(self):
+        urlbar_input = self.browser.navbar.locationbar.urlbar_input
+        self.assertEqual('input', urlbar_input.get_attribute('localName'))
+        self.assertIn('urlbar-input', urlbar_input.get_attribute('className'))
+
+    def test_contextmenu(self):
+        contextmenu = self.browser.navbar.locationbar.contextmenu
+        self.assertEqual('menupopup', contextmenu.get_attribute('localName'))
+
+    def test_contextment_entry(self):
+        contextmenu_entry = self.browser.navbar.locationbar.get_contextmenu_entry('cmd', 'paste')
+        self.assertEqual('cmd_paste', contextmenu_entry.get_attribute('cmd'))
+
+    def test_reloadbutton(self):
+        reload_button = self.browser.navbar.locationbar.reload_button
+        self.assertEqual('Browser:ReloadOrDuplicate', reload_button.get_attribute('command'))
```

Comment:
Please stay with the type of element as for the former property checks. We don't want to fail if internal method names are changing.